src/validation/rules/KnownDirectives.js

```diff
@@ -82,5 +91,18 @@ function getLocationForAppliedNode(appliedTo) {
     case FRAGMENT_SPREAD: return DirectiveLocation.FRAGMENT_SPREAD;
     case INLINE_FRAGMENT: return DirectiveLocation.INLINE_FRAGMENT;
     case FRAGMENT_DEFINITION: return DirectiveLocation.FRAGMENT_DEFINITION;
+    case SCALAR_TYPE_DEFINITION: return DirectiveLocation.SCALAR;
+    case OBJECT_TYPE_DEFINITION: return DirectiveLocation.OBJECT;
+    case FIELD_DEFINITION: return DirectiveLocation.FIELD_DEFINITION;
+    case INTERFACE_TYPE_DEFINITION: return DirectiveLocation.INTERFACE;
+    case UNION_TYPE_DEFINITION: return DirectiveLocation.UNION;
+    case ENUM_TYPE_DEFINITION: return DirectiveLocation.ENUM;
+    case ENUM_VALUE_DEFINITION: return DirectiveLocation.ENUM_VALUE;
+    case INPUT_OBJECT_TYPE_DEFINITION: return DirectiveLocation.INPUT_OBJECT;
+    case INPUT_VALUE_DEFINITION:
+      const parentNode = ancestors[ancestors.length - 3];
```

Comment:
The parser all but guarantees it. The only way this could end up being negative is if you assemble an invalid AST to provide to the validator, and then you probably have other problems on your hands.